lib/use_cases/distribute/overflow_to_developers_channel.rb

```diff
@@ -0,0 +1,44 @@
+module UseCases
+  module Distribute
+    class OverflowToDevelopersChannel
+      def execute(application_prs_by_team:, overflow_at: 15)
+        @application_prs_by_team = application_prs_by_team
+        @overflow_at = overflow_at
+
+        redistribution = []
+
+        application_prs_by_team.each do |application_prs_for_team|
+          redistribution.concat(redistribute(application_prs_for_team[:applications]))
+        end
+
+        if redistribution.any?
+          ensure_govuk_developers_exists
+          govuk_developers_item[:applications].concat(redistribution)
+        end
+
+        application_prs_by_team
+      end
+
+    private
+
+      attr_reader :application_prs_by_team, :overflow_at
+
+      def ensure_govuk_developers_exists
+        return if application_prs_by_team.any? { |h| h[:team_name] == "govuk-developers" }
```

Comment:
This could be `return if govuk_developers_item`. It would return `nil` instead of `false` and the hash instead of `true`, but for what we need it here, the result would be the same.
```
[5] pry(main)> application_prs_by_team = []
=> []
[6] pry(main)> application_prs_by_team.any? { |h| h[:team_name] == "govuk-developers" }
=> false
[7] pry(main)> application_prs_by_team.find { |h| h[:team_name] == "govuk-developers" }
=> nil
[8] pry(main)> application_prs_by_team << { team_name: "govuk-developers", applications: [] }
=> [{:team_name=>"govuk-developers", :applications=>[]}]
[9] pry(main)> application_prs_by_team.any? { |h| h[:team_name] == "govuk-developers" }
=> true
[10] pry(main)> application_prs_by_team.find { |h| h[:team_name] == "govuk-developers" }
=> {:team_name=>"govuk-developers", :applications=>[]}
```

Reply:
Or even
```
unless govuk_developers_item
  application_prs_by_team << { team_name: "govuk-developers", applications: [] }
end
```

Reply:
👍 good idea - changed in https://github.com/alphagov/govuk-dependencies/pull/79/commits/843b01ae448886e6fa288ba727b2e9712496a417 - I changed the name of the method as well since `_item` wasn't the best name in the world.